Modules/Relationships/src/Relationships.Application/Relationships/Commands/CreateRelationship/Handler.cs

```diff
@@ -65,17 +65,29 @@ private async Task CreateAndSaveRelationship()
             _cancellationToken
         );
 
+        var relationships = existingRelationships.ToList();
+
+        EnsureNoTerminatedRelationshipExists(relationships);
```

Comment:
This is domain knowledge. So you should check this in the constructor of the Relationship. Just like all the other checks (e.g. for an existing relationship - which, btw, should just be extended instead of implementing a completely new check)

Reply:
I think I like `EnsureNotStatus`. Even though the name sucks, but I can't find something better either. ;)

Reply:
Yes, the name sucks aesthetically but it works :)

Changes made.